FD/Finite Differencing.ipynb

```diff
@@ -183,15 +182,25 @@
     "    rhs = apply_boundary_conditions(rhs)\n",
     "    return rhs\n",
     "\n",
-    "def rk2_step(u, dx, dt, nu):\n",
-    "    \"\"\" Perform one time step using RK2 integration method. \"\"\"\n",
+    "def rk4_step(u, dx, dt, nu):\n",
```

Comment:
I think that this function is too long . It is more convenient to split them into many functions and just call them here .